test/tests/convergence/reference_residual/tests

```diff
@@ -0,0 +1,11 @@
+[Tests]
+  design = 'ReferenceResidualConvergence.md'
+  issues = '#25931'
+  [reference_residual]
+    type = 'Exodiff'
```

Comment:
I added more such tests, not sure they follow directly what you suggest cause I used `cli_args` instead but it can be easily adjusted. We might need to do voice over this.